### PR TITLE
[2.8] update tests to use the new error assertion helper

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Branch? | 2.8 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: phpunit [hunk 1]
```diff
@@ -56,7 +56,7 @@ if (!file_exists("$PHPUNIT_DIR/phpunit-$PHPUNIT_VERSION/phpunit") || md5_file(__
     if (5.1 <= $PHPUNIT_VERSION && $PHPUNIT_VERSION < 5.4) {
         passthru("$COMPOSER require --no-update phpunit/phpunit-mock-objects \"~3.1.0\"");
     }
-    passthru("$COMPOSER require --dev --no-update symfony/phpunit-bridge \">=3.1@dev\"");
+    passthru("$COMPOSER require --dev --no-update symfony/phpunit-bridge \">=3.2@dev\"");
```
Comment (nicolas-grekas) [inline phpunit]: I think we should make this change into 2.7 to have the same phpunit everywhere

Comment (fabpot) [inline phpunit]: 👍

Comment (nicolas-grekas) [inline phpunit]: done in #19070

File: src/Symfony/Component/Yaml/Tests/ParserTest.php [hunk 2]
```diff
@@ -930,27 +931,12 @@ public function testFloatKeys()
...
-        $deprecations = array();
-        set_error_handler(function ($type, $msg) use (&$deprecations) {
-            if (E_USER_DEPRECATED !== $type) {
-                restore_error_handler();
-
-                return call_user_func_array('PHPUnit_Util_ErrorHandler::handleError', func_get_args());
-            }
-
-            $deprecations[] = $msg;
+            $this->parser->parse($yaml);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Yaml/Tests/ParserTest.php]: $this in closure => not 5.3 compatible. Add `use ($parser)` to the closure?

Comment (xabbuh) [inline src/Symfony/Component/Yaml/Tests/ParserTest.php]: good catch, fixed

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**